src/Raven.Server/ServerWide/Commands/Indexes/PutIndexCommand.cs

```diff
@@ -38,15 +40,11 @@ public PutIndexCommand(IndexDefinition definition, string databaseName, string s
 
         public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
         {
+            var indexValidator = new StaticIndexNameValidator(record);
+            indexValidator.Validate(Definition);
+
             try
             {
-                var indexNames = record.Indexes.Select(x => x.Value.Name).ToHashSet(StringComparer.OrdinalIgnoreCase);
-
-                if (indexNames.Add(Definition.Name) == false && record.Indexes.TryGetValue(Definition.Name, out _) == false)
-                {
-                    throw new InvalidOperationException($"Can not add index: {Definition.Name} because an index with the same name but different casing already exist");
```

Comment:
you will throw different ex now

Reply:
We rethrow it as `RachisApplyException`.
Changed to the expected exception.